tsdb/engine/tsm1/encoding.go

```diff
@@ -289,14 +331,18 @@ func DecodeFloatBlock(block []byte, tdec *TimeDecoder, vdec *FloatDecoder, a *[]
 		return nil, err
 	}
 
+	tdec := timeDecoderPool.Get(0).(*TimeDecoder)
+	vdec := floatDecoderPool.Get(0).(*FloatDecoder)
+
+	var i int
 	// Setup our timestamp and value decoders
 	tdec.Init(tb)
-	if err := vdec.SetBytes(vb); err != nil {
-		return nil, err
+	err = vdec.SetBytes(vb)
+	if err != nil {
+		goto cleanup
```

Comment:
Can we do an inner anonymous function to ensure pool release occurs instead of `goto`?